### DISCUSSION TITLE
Feature Request: Modify CSV export for Tables and Pivot Tables to more closely resembled rendered data in these charts

### DISCUSSION THREAD

**Discussion by SakshiTharkude** [apache/superset#19705] [Ideas]
Charts
Creating charts, selecting a particular visualization type , building queries
For eg: If the selected visualization type is PIVOT TABLE ,
And I want to export the pivot table to a CSV file
It gets exported to a normal csv file
It should to exported in the format of PIVOT
If I create a pivot table of 3*3 matrix the csv file should  give the same output

**Comment by srinify** [comment]
Hey @SakshiTharkude the CSV export actually exports the data _underlying_ a chart. It's basically what's returned from the database :) 

This is consistent and makes sense for all charts but I could see why it's confusing for Table and Pivot Table charts, where the "chart" closely resembles a tabular dataset (in which case, you may expect CSV to export that instead).

I'm moving this to discussions as this is a cool feature request but not a bug per say, as this is working as intended!

**Reply by SakshiTharkude** [reply]
Yes surely not a bug my bad
But you know would look more user friendly if you add such feature only to the pivot's

**Reply by kasiazjc** [reply]
Hi! In pivot table we currently have two ways to export csv - the original file (raw data) and pivoted csv (which more closely resembles what you see on the chart). Attaching the screenshot. Hope this helps.  

We are currently looking to introduce this feature in other charts and improve the whole thing, so there should be some updates soon! 

![image](https://user-images.githubusercontent.com/36897697/163532655-1c194821-39cd-42d7-b226-fe1aebaacb8d.jpeg)

**Reply by SakshiTharkude** [reply]
Its great to hear that !
Thankyou for accepting the request.

**Reply by manojrustagi79** [reply]
> Hi! In pivot table we currently have two ways to export csv - the original file (raw data) and pivoted csv (which more closely resembles what you see on the chart). Attaching the screenshot. Hope this helps.
> 
> We are currently looking to introduce this feature in other charts and improve the whole thing, so there should be some updates soon!
> 
> ![image](https://user-images.githubusercontent.com/36897697/163532655-1c194821-39cd-42d7-b226-fe1aebaacb8d.jpeg)

I am not getting above mentioned option in my preset version. Please let me know how enable those options.

**Reply by nguyenluongky** [reply]
Hi, I tried with version 2.0.0 and saw the options in edit mode, but in the dashboard view how can we enable the option export to pivoted .csv to end user?

**Reply by SakshiTharkude** [reply]
The option is available on the dashboard itself 
![Screenshot (132)](https://user-images.githubusercontent.com/86605906/186381570-e555a645-c364-4e81-8aff-d1803c812633.png)

**Reply by rwibowo** [reply]
@SakshiTharkude which Superset version are you using ?

I am using latest Docker superset image, but I still can't see menu "Download Pivoted CSV" from Dashboard.

**Reply by adimyth** [reply]
The "Export CSV" option in the screenshot attached by @SakshiTharkude downloads the actual underlying data & not the pivoted table. This option I see is currently only available in the pivot table charts. But the same isn't available when these charts are part of a dashboard.

I would also like to have the capability to `"Export as pivoted .CSV"` from within the dashboard.

**Comment by davidburke** [comment]
When exporting to CSV with the 'Pivoted' option, all row fields are concatenated into one column in Excel. Is this a known issue?
![Screen Shot 2022-10-06 at 2 17 01 PM](https://user-images.githubusercontent.com/8650970/194388722-160b7f23-748d-422a-8a34-239cfbcaf30f.png)

**Reply by ekowdd89** [reply]
https://www.youtube.com/watch?v=w0JmD-TlNrk

**Comment by ghost** [comment]
Are there any updates on having pivoted export to csv available in dashboard view? @kasiazjc

**Comment by adimyth** [comment]
@srinify @rwibowo any updates on this one? Is this option still restricted to charts? How can we have these additional options when exporting pivot tables from the dashboard?

**Reply by mtthsbrr** [reply]
I looks like 'Export to pivoted .CSV' is not showing up in the dashboard view because of this condition in the code:

https://github.com/apache/superset/blob/0d0a81c0d2a3efcfa92c7a1ac441760d5a4bc8ff/superset-frontend/src/explore/components/useExploreAdditionalActionsMenu/index.jsx#L324

Could anyone confirm this and maybe even provide a workaround or fix?

**Comment by RavshanbekXB** [comment]
Are there any updates on having pivoted export to csv/excel  available in dashboard view? @kasiazjc

**Comment by fatalitylive** [comment]
hope can export the pivot table2, thank you.

**Comment by adimyth** [comment]
@kasiazjc @rusackas Do we have plans to enable the `Download Pivoted Table` at the dashboard level as well for the `Pivot Table 2` charts? Currently, this works at the chart level in edit mode, but once it's embedded in the dashboard, it's no longer available.

**Reply by rusackas** [reply]
Nope, though it would be helpful. It's just one of many things competing for time and attention, so please feel free to open a PR if it would help!

**Reply by ekowdd89** [reply]
pivot table extension 
https://www.youtube.com/watch?v=w0JmD-TlNrk

**Reply by AlexAT25** [reply]
Is it really works? Anybody tried?
